File: pinyougouparent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//二、引入控制器
app.controller("brandController", function ($scope, $controller, brandService) {
    //引入子控制器
    $controller("baseController", {$scope: $scope});
    //第1步.查询品牌列表
    $scope.findAll = function () {
        //请求地址。相当于ajax请求
        brandService.findAll().success(
            function (respose) {
                $scope.list = respose;
            }
        );
    };
    //################baseController.js###################
    //实现分页
    $scope.findPage = function (pageNum, pageSize) {
        //发送请求并传入当前页数和每页显示的条数
        brandService.findPage(pageNum, pageSize).success(
            function (response) {
                $scope.list = response.rows;//显示当前页展示数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    };
    //########分页结束#########
    //3.保存商品品牌实体类
    $scope.saveBrand = function () {
        var object = null;//定义对象
        if ($scope.entity.id != null) {
            object = brandService.updateBrand($scope.entity);//添加操作
        } else {
            object = brandService.addBrand($scope.entity);//修改操作
        }
        object.success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();//保存成功，刷新品牌列表页面
                } else {
                    alert(response.message);//保存失败
                }
            }
        );
    };
    //4。 //################baseController.js###################
    // 4.1#########修改开始###########
    $scope.findOne = function (id) {
        //发送请求
        brandService.findOne(id).success(
            //接受传来的数据
            function (response) {
                $scope.entity = response;
            }
        );
    };
    //4。#########修改结束###########
    //4.2删除
    $scope.delBrand = function () {
        brandService.delBrand($scope.selectIds).success(
            function (response) {
                if (response.success) {
                    if (confirm("您确定要删除吗？")) {
                        $scope.reloadList();
                    }
                    alert(response.message);//删除成功，刷新品牌列表页面
                } else {
                    alert(response.message);//删除失败
                }
            }
        );
    };
    //5.条件查询：
    $scope.searchEntity = {};//
    $scope.search = function (pageNum, pageSize) {
        //发送请求并传入当前页数和每页显示的条数
        brandService.search(pageNum, pageSize, $scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;//显示当前页展示数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    };
});

File: pinyougouparent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 1.返回品牌列表
     *
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    /**
     * 2.使用分页助手实现品牌分页
     *
     * @param pageNum  当前页码
     * @param pageSize 每页显示条数
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int pageNum, int pageSize) {
        return brandService.findPage(pageNum, pageSize);
    }

    /**
     * 3.添加实体类
     *
     * @return
     */
    @RequestMapping("/addBrand")
    public Result addBrand(@RequestBody TbBrand brand) {
        try {
            brandService.addBrand(brand);//添加
            return new Result(true, "保存成功...");   //判断
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "保存失败...");
        }

    }

    /**
     * 4.
     * ###########修改开始############
     * 第一步：根据id先把TbBrand查询出来
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }
    /**
     * ############修改结束#########
     *第二步：4.2修改之后再保存
     * @param brand
     */
    @RequestMapping("/updateBrand")
    public Result updateBrand(@RequestBody TbBrand brand){
        try {
            brandService.updateBrand(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    /**
     * 5.根据id删除
     * @param ids//复选删除
     */
    @RequestMapping("/delBrand")
    public Result delBrand(Long[] ids){
        try {
            brandService.delBrand(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
    /**
     * 6.品牌分页,品牌条件查询
     *
     * @param brand    品牌实体
     * @param pageNum  当前页码
     * @param pageSize 每页条数
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,int pageNum, int pageSize){
        return brandService.findPage(brand,pageNum,pageSize);
    }

    /**
     * 7.下拉品牌列表
     * @return
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }
}


